### DISCUSSION TITLE
how to compile a real icub app other than a icub sim one.

### DISCUSSION THREAD

**Discussion by jimingda** [robotology/community#619] [Q&A] [answered]
Sorry to bother you, i have a problem here that i can run the  simulation below on icubsim, but i forget how to modify the code to compile a real icub application. Here is the app:
https://github.com/vvv-school/tutorial_gaze-interface

**Comment by jimingda** [comment]
I found the code below just now, change icubSim to icub can the code be compiled into real icub app.

optJoint.put("remote","/icubSim/left_arm");

https://github.com/vvv-school/tutorial_joint-interface/blob/master/src/position.cpp

**Reply by pattacini** [reply]
Here, you don't aim to control the arms of the robot.
For gazing, the module is just fine.

**Comment by pattacini** [comment, marked as answer]
Hi @jimingda 

The tutorial `tutorial_gaze-interface` is different from `tutorial_cartesian-inteface` in that it cannot run on the physical robot because of the intrinsic limitations of the visual pipeline.

The tutorial consists of [3 modules](https://github.com/vvv-school/tutorial_gaze-interface/tree/master/src):
- the **tracker**, which can actually run on the real iCub as it is.
- the **mover**, which is responsible for moving the ball within the simulator (thus, you won't need it in a real settings).
- the **detector**, representing the visual pipeline that is very naive per se and won't be definitely working in a real scenario.

Something close enough to this tutorial is the "famous" [Red Ball demo](https://github.com/robotology/icub-basic-demos/tree/master/demoRedBall) we usually showcase once we deliver the robot.
